### ISSUE TITLE
Driver and upper computer ?

### ISSUE THREAD

**Issue by Roc-0** [davidgfnet/stm32-dfu-bootloader#3]
After I loaded the program onto the chip, the following problems appeared on the computer side:

![image](https://user-images.githubusercontent.com/32725337/86340848-1b6e2980-bc88-11ea-8329-b50c06b226e0.png)

I want to confirm with you and ask the following questions:
1. Is the upper computer software "dfu-util"?
2. Where can I get the driver for win10 64-bit system?

**Comment by davidgfnet** [comment]
Hey there!
I'm on Linux, so cannot really comment on Windows. Have you tried the STM32 DFU util for windows?

**Comment by oldgerman** [comment]
I used f103c8t6，same problem with win10😥
![20200815195211](https://user-images.githubusercontent.com/43802352/90311727-d26cee80-df30-11ea-9c53-4de4a0a6cd4f.png)
DfuSe Demo can't recognize the device：
![20200815195454](https://user-images.githubusercontent.com/43802352/90311751-31cafe80-df31-11ea-98ab-223e286d5ba7.png)
But DfuSe can recogSnize F303 normally.

**Comment by jamesle-0812** [comment]
Hi, I got same problem here.
![image](https://user-images.githubusercontent.com/38745566/118347425-7d1c4700-b56d-11eb-9faa-2dd37173b716.png)
DfuSe Demo can't get the device as well!

**Comment by davidgfnet** [comment]
Hey there!
Sorry to hear. Not very familiar with these tools.
I tried with one windows VM today, I got the same results. However I installed Zadig (https://zadig.akeo.ie/) and used WebUSB with a DFU util (I used this one: https://devanlai.github.io/webdfu/dfu-util/) and I managed to Download some firmware!
Not sure if the Dfuse Demo works only for a subset of DFU devices or something...

**Comment by diggit** [comment]
Windows need extra USB descriptor which tells them, they should attach winusb or similar driver.
https://learn.microsoft.com/en-us/windows-hardware/drivers/usbcon/microsoft-defined-usb-descriptors
https://github.com/newaetech/naeusb/blob/main/wcid.md

**Comment by davidgfnet** [comment]
Thanks for the hint, lemme take a look at it and see what I can do :)

**Comment by davidgfnet** [comment]
Just added WINUSB support. Seems to work OK on Win10 for me. Can you verify?

**Comment by askyuan** [comment]
zadig will be get the driver

**Comment by diggit** [comment]
> zadig will be get the driver

Yeah, zadig can assign any driver to USB device. Point is to avoid need of zadig or similar tools and have Windows do it automatically. It can be done using recently implemented wicd descriptor. Zadig should not be necessary for this BL anymore.

**Comment by davidgfnet** [comment]
Thanks @diggit . Has anyone confirmed it worked for you? I'm not using Windows, just for testing this Issue. However Im not sure what tools you use. I tried DsFuse Demo but I think it expects some specific VID/PID device (not just "any")?
